Source/WebKit/NetworkProcess/PrivateClickMeasurement/PrivateClickMeasurementEphemeralStore.cpp

```diff
@@ -0,0 +1,145 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "PrivateClickMeasurementEphemeralStore.h"
+
+#include <WebCore/PrivateClickMeasurement.h>
+
+namespace WebKit::PCM {
+
+EphemeralStore::EphemeralStore() = default;
+EphemeralStore::~EphemeralStore() = default;
+
+void EphemeralStore::insertPrivateClickMeasurement(WebCore::PrivateClickMeasurement&& attribution, PrivateClickMeasurementAttributionType type, CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(attribution.isEphemeral() == WebCore::PCM::AttributionEphemeral::Yes);
+    ASSERT_UNUSED(type, type == PrivateClickMeasurementAttributionType::Unattributed);
+    m_clickMeasurement = WTFMove(attribution);
+    completionHandler();
+}
+
+void EphemeralStore::markAllUnattributedPrivateClickMeasurementAsExpiredForTesting()
+{
+    if (m_clickMeasurement && !m_clickMeasurement->attributionTriggerData())
+        reset();
+}
+
+void EphemeralStore::attributePrivateClickMeasurement(WebCore::PCM::SourceSite&& sourceSite, WebCore::PCM::AttributionDestinationSite&& destinationSite, const ApplicationBundleIdentifier& applicationBundleIdentifier, WebCore::PCM::AttributionTriggerData&& attributionTriggerData, WebCore::PrivateClickMeasurement::IsRunningLayoutTest isLayoutTest, CompletionHandler<void(std::optional<WebCore::PCM::AttributionSecondsUntilSendData>&&, DebugInfo&&)>&& completionHandler)
+{
+    if (!m_clickMeasurement)
+        return completionHandler(std::nullopt, { });
+
+    if (m_clickMeasurement->sourceSite() != sourceSite || m_clickMeasurement->destinationSite() != destinationSite)
+        return completionHandler(std::nullopt, { });
+
+    if (!applicationBundleIdentifier.isEmpty() && m_clickMeasurement->sourceApplicationBundleID() != applicationBundleIdentifier)
+        return completionHandler(std::nullopt, { });
+
+    completionHandler(m_clickMeasurement->attributeAndGetEarliestTimeToSend(WTFMove(attributionTriggerData), isLayoutTest), { });
+}
+
+void EphemeralStore::privateClickMeasurementToStringForTesting(CompletionHandler<void(String)>&& completionHandler) const
+{
+    if (!m_clickMeasurement)
+        return completionHandler("\nNo stored Private Click Measurement data.\n"_s);
```

Comment:
Maybe add 'ephemeral' in this log statement to differentiate it from the persistent one.